code/datums/components/examine_balloon.dm

```diff
@@ -0,0 +1,74 @@
+/// Adds clickable balloons whenever someone holds the examine key (is it still shift in the future?)
+/datum/component/examine_balloon
+	/// Offset applied on the hologram
+	var/pixel_y_offset
+	/// Our x and y size is multiplied by this, for small sprites like buttons
+	var/size_upscaling
+
+	/// Offset applied on the bubble
+	var/pixel_y_offset_arrow = 16
+	/// The alpha we apply to the hologram
+	var/hologram_alpha = 200
+	/// Add a hologram when we're in these directions
+	var/draw_in_dirs = NORTH | EAST | WEST
+
+/datum/component/examine_balloon/Initialize(pixel_y_offset = 28, pixel_y_offset_arrow = 16, size_upscaling = 1)
+	. = ..()
+
+	if(!ismovable(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.pixel_y_offset = pixel_y_offset
+	src.size_upscaling = size_upscaling
+	src.pixel_y_offset_arrow = pixel_y_offset_arrow
+
+	var/atom/atom_parent = parent
+
+	RegisterSignal(atom_parent, COMSIG_ATOM_UPDATE_ICON, PROC_REF(on_update_icon))
+	// We use UPDATED_ICON specifically because we need to be last in the icon chain, even if UPDATE_OVERLAYS would otherwise be more apt
+	RegisterSignal(atom_parent, COMSIG_ATOM_UPDATED_ICON, PROC_REF(on_updated_icon))
+
+	atom_parent.update_icon(UPDATE_OVERLAYS)
+
+/datum/component/examine_balloon/proc/on_update_icon(atom/movable/parent, updates)
+	SIGNAL_HANDLER
+
+	if(!(updates & UPDATE_OVERLAYS))
+		return
+
+	parent.overlays.Cut() //Clear all previous overlays, so we don't infinitely stack with our own overlays
+	parent.managed_overlays.Cut()
```

Comment:
this isn't valid, more things then you might try to use the overlays list. either here or in `on_updated_icon` you should just `cut_overlay(hologram) // same thing for the arrow`

Reply:
not resolved

Reply:
also you both should not be and have no reason to touch managed_overlays (unless I have forgotten something)

Reply:
cut_overlays() does not clear the managed overlays, meaning that on next icon update it compares the new overlays with managed overlays, and if they're equal it doesn't update the icon. In effect this means wallmounts don't have overlays if they call update_icons without having a different appearance

also I see now that I completely misread what you said. I did try that but it didn't work but I tried again and now it works so ???? 🎉🎉🎉